source/main.cc

```diff
@@ -616,6 +620,16 @@ run_simulator(const std::string &raw_input_as_string,
 
           std::ofstream file(output_directory + "original.prm");
           file << raw_input_as_string;
+
+          // If using the Geodynamic World Builder, create output/original.wb
+          // containing the exact file used to create the world:
+          std::string world_builder_file = prm.get("World builder file");
+          if (world_builder_file != "")
+            {
+              std::stringstream json_input_stream(WorldBuilder::Utilities::read_and_distribute_file_content(world_builder_file));
```

Comment:
This seems a bit overly complicated to me. On line 611 you make sure you only run this on process zero. So there is no reason to read and distribute. I would just read the file with an ifstream and write it out with a ofstream. Something like this:

```
    std::ifstream  wb_source(world_builder_file, std::ios::binary);
    std::ofstream  wb_destination(output_directory + "original.wb",   std::ios::binary);

    wb_destination << src.rdbuf();
```